core/base_abstractions/sensor.py

```diff
@@ -158,23 +166,34 @@ def _get_observation_space(self) -> gym.spaces.Tuple:
         only if the expert failed to generate a true expert action. The
         value `num actions in task` should be in `config["nactions"]`
         """
-        return gym.spaces.Tuple(
-            (gym.spaces.Discrete(self.nactions), gym.spaces.Discrete(2))
-        )
+        return su.flatten_space(self.unflattened_observation_space)
 
     def get_observation(
         self, env: EnvType, task: SubTaskType, *args: Any, **kwargs: Any
     ) -> Any:
         # If the task is completed, we needn't (perhaps can't) find the expert
         # action from the (current) terminal state.
         if task.is_done():
-            return np.array([-1, False], dtype=np.int64)
+            return np.array([self.action_space.sample(), False])
```

Comment:
Can we add a function like:
```python
@lazy_property
def _zeroed_action(self):
    return self.action_space.sample().zero_()
```
and then change this line to `return np.array([self._empty_action, False])`?

Reply:
Ok.